components/layout_2020/flow/float.rs

```diff
@@ -84,66 +86,66 @@ pub(crate) struct PlacementAmongFloats<'a> {
     /// The next band, needed to know the height of the last band in current_bands.
     next_band: FloatBand,
     /// The size of the object to place.
-    object_size: LogicalVec2<Length>,
+    object_size: LogicalVec2<Au>,
     /// The minimum position in the block direction for the placement. Objects should not
     /// be placed before this point.
-    ceiling: Length,
+    ceiling: Au,
     /// The inline position where the object would be if there were no floats. The object
     /// can be placed after it due to floats, but not before it.
-    min_inline_start: Length,
+    min_inline_start: Au,
     /// The maximum inline position that the object can attain when avoiding floats.
-    max_inline_end: Length,
+    max_inline_end: Au,
 }
 
 impl<'a> PlacementAmongFloats<'a> {
     pub(crate) fn new(
         float_context: &'a FloatContext,
-        ceiling: Length,
-        object_size: LogicalVec2<Length>,
+        ceiling: Au,
+        object_size: LogicalVec2<Au>,
         pbm: &PaddingBorderMargin,
     ) -> Self {
-        assert!(!ceiling.px().is_infinite());
-        let mut current_band = float_context.bands.find(ceiling).unwrap();
-        current_band.top = ceiling;
+        assert!(ceiling.to_f32_px() != INFINITY || ceiling.to_f32_px() != NEG_INFINITY);
```

Comment:
Use `&&` instead of `||`?

Anyways, this makes me realize that most of these `is_infinite()` were actually meant to just test for positive infinity.